### DISCUSSION TITLE
Stackdriver: Add support for select_slo_burn_rate(SLO_NAME, LOOKBACK_PERIOD)

### DISCUSSION THREAD

**Discussion by tsekine354** [grafana/grafana#41824] [Data Sources]
**What would you like to be added**:
Add `SLI Burn Rate - select_slo_burn_rate (SLO_NAME, LOOKBACK_PERIOD)` as a Stackdriver SLO query Selector option. 

It's mentioned in https://github.com/grafana/grafana/issues/22627, but dropped from initial PR https://github.com/grafana/grafana/pull/22917 as discussed in the issue.

**Why is this needed**:
`select_slo_burn_rate(SLO_NAME, LOOKBACK_PERIOD)` is one of the most useful SLO related timeseries generator function. Or, it's one of the most important function, as it's used for alerting  **on Stackdriver**. It's always necessary to see how source data behaves to set up alerting.

BTW, If `select_slo_burn_rate()` is supported by grafana, SLO alerting could be set up **on grafana**, but it's a just side-effect of this enhancement request for *me*.

**Comment by sunker** [comment]
Sounds like a good idea to me. @Dnefedkin what do you think?

**Comment by ghost** [comment]
Agree, with @tsekine354 ,  it's  beneficial to have an ability visualize SLO burn rate with different lookback periods (for fast burn and slow burn alerts).

**Comment by sunker** [comment]
For the lookback period, would it make sense to let the default value be calculated based on the dashboard time range? I.e if the time range is the last 6 hours, then it would be 21600s? Then this value could be overridden with a custom period as well.

**Comment by ghost** [comment]
Lookback period longer than 24 hours is not supported, so calculating it based on dashboard time range might lead to errors while calling select_slo_burn_rate function in the background.

@steventhurgood, @knyar - as authors of https://sre.google/workbook/alerting-on-slos/ , what do you think about default value of a lookback period that we can use?

**Comment by tsekine354** [comment]
I usually tell my colleagues to look at https://cloud.google.com/stackdriver/docs/solutions/slo-monitoring/alerting-on-budget-burn-rate for the burn rate. That document suggests:
- 1 or 2 hours for fast burn rate
- 24 hours for slow burn rate

fyi.

**Comment by andresmgot** [comment]
For the record, this was added here https://github.com/grafana/grafana/pull/53710 and available since Grafana 9.1.1. I am going to archive this discussion but open a new issue if you find any trouble.
